### PR TITLE
[sc-5092] spinner handle interrupt signal

### PR DESCRIPTION
**Pull request by steb-veksler**

### CODE REVIEW

File: pkg/auth/device_code.go [hunk 1]
```diff
@@ -54,15 +58,30 @@ func (deviceCode *DeviceCode) PollToken(auth0Token *Auth0Token) error {
 	data.Set("device_code", deviceCode.DeviceCode)
 	data.Set("grant_type", "urn:ietf:params:oauth:grant-type:device_code")
 
-	err := spinner.Poll(func() (bool, error) {
-		return fetchTokenFunc(auth0Token, data)
-	},
-		DEVICE_CODE_POLLING_INTERVAL,
-		DEVICE_CODE_POLLING_TIMEOUT)
+	fetchTokenFunc := func() (bool, error) {
```
Comment (noamApps) [inline pkg/auth/device_code.go:61]: why not keeping fetchTokenFunc like it was? I think it was cleaner to have it separated
(maybe Im missing something but looks like only Poll really changed with the new ctx arg

File: cmd/status.go [hunk 2]
```diff
@@ -165,20 +165,19 @@ func waitForAlligators(ctx context.Context, kubeClient *k8s.Client, helmRelease
...
-				return false, err
-			}
 
-			spinner.Message(fmt.Sprintf(WAIT_FOR_ALLIGATORS_FORMAT, runningAlligators, expectedAlligatorsCount))
-			return runningAlligators == expectedAlligatorsCount, nil
-		},
-		PODS_POLLING_INTERVAL,
-		ALLIGATORS_POLLING_TIMEOUT,
-	)
+	isAlligatorRunningFunc := func() (bool, error) {
```
Comment (noamApps) [inline cmd/status.go:169]: just to make sure, other then separating this function to a separate declaration, nothing changed other by adding context right?

File: pkg/auth/device_code.go [hunk 3]
```diff
@@ -54,15 +58,30 @@ func (deviceCode *DeviceCode) PollToken(auth0Token *Auth0Token) error {
...
+		return false, err
+	}
+
+	err = spinner.Poll(ctx, fetchTokenFunc, DEVICE_CODE_POLLING_INTERVAL, DEVICE_CODE_POLLING_TIMEOUT)
+
 	if err == nil {
 		return nil
 	}
 
+	spinner.StopFail()
```
Comment (noamApps) [inline pkg/auth/device_code.go:83]: it might be called twice now - is that ok ? do we still need the stopFail call in the ui.ErrSpinnerTimeout case?

**Review by noamApps — COMMENTED** [review]
so just to make sure , the only real change in order to support ctrl+C is to pass to cobra context and adding a handler to Done state in the select statement?  the cobra context trpas the interrupt?

File: pkg/auth/device_code.go (hunk 1 above)
Comment (steb-veksler) [reply pkg/auth/device_code.go:61]: https://github.com/groundcover-com/cli/blob/1bb7eae10580d6f43c2f9bdc0f5c850a3234b6cc/pkg/auth/device_code.go#L58

you need to have wrap func to be able call the function with parameters, for me less readable

File: cmd/status.go (hunk 2 above)
Comment (steb-veksler) [reply cmd/status.go:169]: right

File: pkg/auth/device_code.go (hunk 3 above)
Comment (steb-veksler) [reply pkg/auth/device_code.go:83]: fixed

**Comment by steb-veksler** [comment]
> so just to make sure , the only real change in order to support ctrl+C is to pass to cobra context and adding a handler to Done state in the select statement? the cobra context trpas the interrupt?

yes, we create context with signal interrupt
https://github.com/groundcover-com/cli/blob/57f7f51c604565691daf06bfc64fb8c93ff99e03/main.go#L47

**Review by noamApps — APPROVED** [review]

**Pull request merged**